docs/modules/ROOT/pages/how-tos/upgrade.adoc

```diff
@@ -0,0 +1,84 @@
+= How to Upgrade cert-manager
+
+== Upgrade cert-manager from 0.x to 1.x
+
+The upgrade is done with the following steps:
+
+. Prepare 3-way merge
+. Build Commodore catalog with updated `component-cert-manager`
+. Ensure ArgoCD sync
+. Verify certificates
+
+=== Prerequisites
+
+. `commodore`
+. `kubectl` or `oc`
+. https://github.com/mikefarah/yq[`yq`] version 4 (alternatively, any editor works)
+
+=== Prepare 3-way merge
+
+This step updates the `"kubectl.kubernetes.io/last-applied-configuration"` annotation which is required for a successful 3-way merge.
+
+[source,bash]
+----
+cat <<'EOF' > cert-manager-prepare-merge-crds.sh
+#!/bin/sh
+
+set -euo pipefail
+
+for crd in certificaterequests.cert-manager.io certificates.cert-manager.io challenges.acme.cert-manager.io clusterissuers.cert-manager.io issuers.cert-manager.io orders.acme.cert-manager.io; do
+  # Get CRD definition in YAML
+  kubectl get crd "${crd}" -o yaml > "${crd}.yaml"
+  # Remove all metadata properties except `metadata.name`
+  yq -i eval 'del(.status) | del(.metadata) | .metadata.name += "'${crd}'"' "${crd}.yaml"
+  # Apply the CRD again (this shouldn't change anything, except adding the annotation "kubectl.kubernetes.io/last-applied-configuration")
+  # You will also see some warnings in the output mentioning the annotation.
+  # This is expected and actually required.
+  kubectl apply -f "${crd}.yaml"
+done
+EOF
+
+sh cert-manager-prepare-merge-crds.sh
+----
+
+=== Build Commodore catalog with updated `component-cert-manager`
+
+[source,bash]
+----
+# Set the cluster you'd like to upgrade
+CLUSTER_ID=XXX
+
+# Build catalog and push changes
+commodore catalog compile $CLUSTER_ID --push -i
+----
+
+=== Ensure ArgoCD sync
+
+Extract the password for the webgui (replace pbcopy with something that works for your OS)
+
+[source,bash]
+----
+kubectl -n syn get secret/steward -o jsonpath='{.data.token}' | base64 --decode | pbcopy
+----
+
+Start a port-forwarding to access the ArgoCD web gui.
+
+[source,bash]
+----
+kubectl -n syn port-forward svc/argocd-server 8080:80
+----
+
+Visit http://localhost:8080/applications/cert-manager and click `🔄 SYNC`.
+
+=== Verify cert-manager and certificates
+
+[source,bash]
+----
+# Verify successful sync and health status in argo
+
+# Check for errors in the logs
+kubectl -nsyn-cert-manager logs -l "app.kubernetes.io/name=cert-manager"
+
+# TODO can we / should we check certificates?
```

Comment:
Can and should we add some more checks?

Reply:
It's an extra optional effort. One would need to have a DNS record and point to the cluster, then we can create an Ingress, then verify if the stack works, then remove again.

We could manually test the upgrade on a "lesser-important" cluster first where wildcard DNS is available. If that works, I'd consider the upgrade successful for all other clusters too.

Reply:
Actually, maybe we should suggest it. It's up to the user whether they do it or not

Reply:
How do we suggest it? Keep it simple and suggest creating a new Ingress or some more guidance?

Reply:
Check out the help page `kubectl create ingress --help`, there are a few rules described how to quickly generate one. The ingress doesn't need to resolve to a service + app, just a valid certificate when doing a `curl` request or so.